third_party/txt/src/txt/directory_asset_data_provider.cc

```diff
@@ -0,0 +1,74 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <dirent.h>
+#include <sstream>
+#include "lib/ftl/logging.h"
+#include "third_party/skia/include/core/SkStream.h"
+#include "third_party/skia/include/core/SkTypeface.h"
+#include "txt/directory_asset_data_provider.h"
+
+namespace txt {
+
+DirectoryAssetDataProvider::DirectoryAssetDataProvider(
+    const std::string& directory_path) {
+  auto directory_closer = [](DIR* directory) {
+    if (directory != nullptr) {
+      ::closedir(directory);
+    }
+  };
+
+  std::unique_ptr<DIR, decltype(directory_closer)> directory(
+      ::opendir(directory_path.c_str()), directory_closer);
+
+  if (directory == nullptr) {
+    return;
+  }
+
+  for (struct dirent* entry = ::readdir(directory.get()); entry != nullptr;
+       entry = ::readdir(directory.get())) {
+    if (entry->d_type != DT_REG) {
+      continue;
+    }
+
+    std::string file_name(entry->d_name, entry->d_namlen);
```

Comment:
Does this build on Linux?  d_namlen is a non-POSIX extension

Reply:
Hmm, you're right. Probably not. I will check on Linux and remove if not present.